src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TagBuilder.cs

```diff
@@ -4,18 +4,19 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.Net;
 using System.Text;
 using Microsoft.AspNet.Mvc.Core;
 using Microsoft.Framework.Internal;
+using Microsoft.Framework.WebEncoders;
 
 namespace Microsoft.AspNet.Mvc.Rendering
 {
     public class TagBuilder
     {
         private string _innerHtml;
+        private IHtmlEncoder _htmlEncoder;
 
-        public TagBuilder(string tagName)
```

Comment:
Should we keep the `TagBuilder(string tagName)` ctor overload for convenience? I'm sure there's plenty of existing code that uses this, and it's a bad experience to force devs to provide it every time.

You could also imagine a `HtmlHelper.CreateTagBuilder(string tagName)` convenience factory that instantiates a TagBuilder with the given name and with any services that the HtmlHelper is able to provide. This allows you to add ctor overloads in the future, and anybody who is calling the CreateTagBuilder convenience method would just get the correct behaviors automatically.